Libplanet/Consensus/Validator.cs

```diff
@@ -0,0 +1,165 @@
+        public Validator(
+            PublicKey publicKey,
+            BigInteger power)
+        {
+            PublicKey = publicKey;
+            Power = power;
+        }
+
+        public Validator(ImmutableArray<byte> marshaled)
+        {
+            var codec = new Codec();
+            Dictionary dict = (Dictionary)codec.Decode(marshaled.ToArray());
+            PublicKey = new PublicKey(dict.GetValue<Binary>(PublicKeyKey).ByteArray);
+            Power = new BigInteger(dict.GetValue<Binary>(PowerKey).ToByteArray());
+        }
+
+        public Validator(byte[] marshaled)
+            : this(marshaled.ToImmutableArray())
+        {
+        }
+
+        public Validator(Binary marshaled)
+            : this(marshaled.ByteArray)
+        {
+        }
+
+        /// <summary>
+        /// A <see cref="PublicKey"/> of validator operator.
+        /// </summary>
+        public PublicKey PublicKey { get; }
+
+        /// <summary>
+        /// A <see cref="Address"/> of validator operator's <see cref="PublicKey"/>.
+        /// </summary>
+        public Address OperatorAddress => PublicKey.ToAddress();
+
+        /// <summary>
+        /// The <see langword="Power"/> of validator.
+        /// </summary>
+        public BigInteger Power { get; }
+
+        /// <summary>
+        /// The marshaled byte array of <see cref="Validator"/>.
+        /// </summary>
+        public ImmutableArray<byte> ByteArray
+        {
+            get
+            {
+                var codec = new Codec();
+                var dict = Dictionary.Empty;
+                dict = dict.Add(PublicKeyKey, PublicKey.Format(true));
+                if (Power is { } power)
+                {
+                    dict = dict.Add(PowerKey, power.ToByteArray());
+                }
+                else
+                {
+                    dict = dict.Add(PowerKey, BigInteger.MinusOne.ToByteArray());
+                }
+
+                return codec.Encode(dict).ToImmutableArray();
+            }
+        }
+
+        public static bool operator ==(Validator obj, Validator other)
+        {
+            return obj.Equals(other);
+        }
+
+        public static bool operator !=(Validator obj, Validator other)
+        {
+            return !(obj == other);
+        }
+
+        public static bool operator ==(Validator obj, PublicKey other)
+        {
+            return obj.Equals(other);
+        }
+
+        public static bool operator !=(Validator obj, PublicKey other)
+        {
+            return !(obj == other);
+        }
+
+        public byte[] ToByteArray()
+        {
+            return ByteArray.ToArray();
+        }
+
+        public override bool Equals(object? obj)
+        {
+            if (obj is Validator other)
+            {
+                return Equals(other);
+            }
+
+            return false;
+        }
+
+        public bool Equals(Validator? other)
+        {
+            return PublicKey.Equals(other?.PublicKey);
```

Comment:
Is not considering power intended? If so, why?

Reply:
What I've thought was, we're often use validator's identity as it's public key.
In practical use, when we compare validator `A` and validator `B`, we often don't pay attention to its powers. What we're interested in is their identity.
So, I've thought judging `Validator("A", 100)` same as `Validator("A", 1000)` would be better for practical use.
I thought, when we're interested in their power, we would call `ValidatorA.Power.Equals(ValidatorB.Power)`.
May people can confused with this notation?

Reply:
Resolved